astropy/stats/lombscargle/tests/test_lombscargle.py

```diff
@@ -357,3 +374,51 @@ def test_autopower(data):
 
     assert_allclose(freq1, freq2)
     assert_allclose(power1, power2)
+
+
+@pytest.mark.parametrize('with_units', [True, False])
+@pytest.mark.parametrize('errors', ['none', 'partial', 'full'])
+@pytest.mark.parametrize('center_data', [True, False])
+@pytest.mark.parametrize('fit_mean', [True, False])
+@pytest.mark.parametrize('nterms', [0, 1, 2])
+def test_model_parameters(data, nterms, fit_mean, center_data,
+                          errors, with_units):
+    if nterms == 0 and not fit_mean:
+        return
+
+    t, y, dy = data
+    frequency = 1.5
+    if with_units:
+        t = t * units.day
+        y = y * units.mag
+        dy = dy * units.mag
+        frequency = frequency / t.unit
+
+    if errors == 'none':
+        dy = None
+    elif errors == 'partial':
+        dy = dy[0]
+    elif errors == 'full':
+        pass
+    else:
+        raise ValueError("Unrecognized error type: '{0}'".format(errors))
+
+    ls = LombScargle(t, y, dy,
+                     nterms=nterms,
+                     fit_mean=fit_mean,
+                     center_data=center_data)
+    tfit = np.linspace(0, 20, 10)
+    if with_units:
+        tfit = tfit * units.day
+
+    model = ls.model(tfit, frequency)
+    params = ls.model_parameters(frequency)
+    design = ls.design_matrix(frequency, t=tfit)
+    offset = ls.offset()
+
+    assert len(params) == int(fit_mean) + 2 * nterms
+
+    print(offset)
+    print(design)
+    print(params)
```

Comment:
These `print` calls should be removed (debugging leftovers?).

Reply:
Yep, thanks.